### ISSUE TITLE
rewrite jolt engine debug renderer for cross-platform

### ISSUE THREAD

**Issue by yangfengzzz** [BoomingTech/Piccolo#366]
I found the debug renderer is used from jolt engine only support DX11. I think it is better to rewrite by using Vulkan to support debug in macos.

**Comment by hyv1001** [comment]
Yes, it is better to implement the jolt renderer with the engine's rendering system. It's on our roadmap.

**Comment by RevoluPowered** [comment]
Dumb question is there an interface for doing this at the moment?

I would not mind if there was some dumb interface like IDebugRenderer we could implement in our game engine?
